backend/infrahub/graphql/types/branch.py

```diff
@@ -46,4 +51,80 @@ async def get_list(
             if not objs:
                 return []
 
-            return [await obj.to_graphql(fields=fields) for obj in objs if obj.name != GLOBAL_BRANCH_NAME]
+            return await cls._map_fields_to_graphql(objs=objs, fields=fields)
+
+
+class RequiredStringValueField(InfrahubObjectType):
+    value = String(required=True)
+
+
+class NonRequiredStringValueField(InfrahubObjectType):
+    value = String(required=False)
+
+
+class NonRequiredBooleanValueField(InfrahubObjectType):
+    value = Boolean(required=False)
+
+
+class StatusField(InfrahubObjectType):
+    value = InfrahubBranchStatus(required=True)
+
+
+class InfrahubBranch(BranchType):
+    id = String(required=True)
+    created_at = String(required=False)
+
+    name = Field(RequiredStringValueField, required=True)
+    description = Field(NonRequiredStringValueField, required=False)
+    origin_branch = Field(NonRequiredStringValueField, required=False)
+    branched_from = Field(NonRequiredStringValueField, required=False)
+    status = Field(StatusField, required=True)
```

Comment:
Here we override `status` from the parent `BranchType` class but we don't override `graph_version`

So a query ends up being:

```graphql
query MyQuery {
  InfrahubBranch {
    count
    edges {
      node {
        graph_version
        status {
          value
        }
        name {
          value
        }
      }
    }
  }
}
```